### PR TITLE
Webcfg coverity test

### PR DESCRIPTION
**Pull request by suvi-123**

### CODE REVIEW

File: src/webcfg_db.c [hunk 1]
```diff
@@ -915,7 +915,9 @@ int process_webcfgdb( webconfig_db_data_t *wd, msgpack_object *obj )
         size_t entries_count = -1;
 
         entries_count = array->size;
+	pthread_mutex_lock (&webconfig_db_mut);
```
Comment (shilpa24balaji) [inline src/webcfg_db.c:918]: Good catch

File: src/webcfg_multipart.c [hunk 2]
```diff
@@ -486,59 +487,68 @@ WEBCFG_STATUS parseMultipartDocument(void *config_data, char *ct , size_t data_s
...
 		while((ptr_count - str_body) < (int)data_size )
 		{
-			ptr_count = memchr(ptr_count, '-', data_size - (ptr_count - str_body));
-			if(0 == memcmp(ptr_count, last_line_boundary, strlen(last_line_boundary)))
-			{
-				num_of_parts++;
-				break;
-			}
-			else if(0 == memcmp(ptr_count, line_boundary, strlen(line_boundary)))
+			if(ptr_count)
```
Comment (shilpa24balaji) [inline src/webcfg_multipart.c:490]: while loop too has condition check on the ptr_count, so there also null check is needed

File: src/webcfg_multipart.c [hunk 3]
```diff
@@ -486,59 +487,68 @@ WEBCFG_STATUS parseMultipartDocument(void *config_data, char *ct , size_t data_s
...
 		while((ptr_lb - str_body) < (int)data_size)
 		{
-			ptr_lb = memchr(ptr_lb, '-', data_size - (ptr_lb - str_body));
-			if(0 == memcmp(ptr_lb, last_line_boundary, strlen(last_line_boundary)))
-			{
-				WebcfgDebug("last line boundary \n");
-				break;
-			}
-			else if(0 == memcmp(ptr_lb, line_boundary, strlen(line_boundary)))
+			if(ptr_lb)
```
Comment (shilpa24balaji) [inline src/webcfg_multipart.c:511]: Add ptr_lb null check in while condition itself

File: src/webcfg_multipart.c [hunk 4]
```diff
@@ -486,59 +487,68 @@ WEBCFG_STATUS parseMultipartDocument(void *config_data, char *ct , size_t data_s
...
 			{
-				ptr_lb = ptr_lb+(strlen(line_boundary))-1;
-				ptr_lb1 = ptr_lb+1;
-				num_of_parts = 1;
-				while(0 != num_of_parts % 2)
+				ptr_lb = memchr(ptr_lb, '-', data_size - (ptr_lb - str_body));
```
Comment (shilpa24balaji) [inline src/webcfg_multipart.c:513]: Since the pointer is getting changed in between we may have to add null check in every if condition and not as a whole

Comment (suvi-123) [reply src/webcfg_multipart.c:513]: I have added null check in the place where  the pointer is getting changed in line number 526

File: src/webcfg_client.c [hunk 5]
```diff
@@ -226,8 +226,14 @@ static void get_parodus_url(char **parodus_url, char **client_url)
 	{
 		char str[255] = {'\0'};
 		//TODO:: Use Fgets to fix coverity issue "fscanf" assumes an arbitrarily long string.
-		while(fscanf(fp,"%s", str) != EOF)
-		{
+		while(fgets(str,255,fp))
```
Comment (shilpa24balaji) [inline src/webcfg_client.c:229]: what is the issue reported on this?

Comment (sadhyama) [reply src/webcfg_client.c:229]: please share device logs for this test

**Pull request closed without merging**